code/modules/mob/living/carbon/carbon.dm

```diff
@@ -48,7 +48,36 @@
 		if(germ_level < GERM_LEVEL_MOVE_CAP && prob(8))
 			germ_level++
 
-
+		if(src.species?.reagent_tag == IS_SLIME) // Slimes clean the floor. Code stolen from the roombas
+			var/turf/tile = loc
+			var/nutrition_gained = 5 // Up here for ease of modification
+			if(isturf(tile))
+				tile.clean_blood()
+				for(var/A in tile)
```

Comment:
this for loop is a little inefficient for something that will run every single time a slime moves, it could make a lag machine pretty easily

Maybe setting up a signal proc would be better? So here it would send

SEND_SIGNAL(tile, COMSIG_CLEAN, src)

And that would link up to procs defined by using RegisterSignal for functions objects, effects and creatures within that tile.

Reply:
1. I have no idea how the heck signals works.
2. It is the same code that the roombas use to clean the tile.